industry_benchmarks/utils/data_gathering.py

```
@@ -444,14 +445,224 @@ def gather_ligand_scores(
+        if reversed_path_parts[idx] == reversed_path_parts[idx + 1]:
+            break
+    # no dupes
+    else:
+        print("Path didn't have any dupes")
+        return path
+
+    del reversed_path_parts[idx]
+    return pathlib.Path(*reversed(reversed_path_parts))
+
+
+def find_data_folder(result: dict) -> None | pathlib.Path:
+    """
+    Find the path to the cleaned up results file for this transformation result.
+    """
+    # get the name of the key which is a gufe token
+    # for the only ProtocolUnitResult-* in unit_results
+    # this means we can grab the first that matches since there is only
+    # one ProtocolUnitResult-*
+    proto_key = next(
+        k
+        for k in result["unit_results"].keys()
+        if k.startswith("ProtocolUnitResult")
+    )
+    results_dir = (
+        pathlib.Path(result["unit_results"][proto_key]["outputs"]["nc"]["path"])
+        .resolve()
+        .parent
+    )
+    # if the dir doesn't exist, we should try and fix it
+    if not results_dir.is_dir():
+        print("Fixing path to results dir")
+        # Depending on the relative location to the result dir, we might have
+        # to fix a duplicate folder, see this post for more details
+        # https://github.com/OpenFreeEnergy/IndustryBenchmarks2024/pull/83#discussion_r1689003616
+        results_dir = remove_first_reversed_sequential_duplicate_from_path(
+            results_dir
+        )
+        # Now we should check if the dir exists
+        if not results_dir.is_dir():
+            print("Can't find results directory, skipping")
+            return None
+
+    # now check that all of the results files can be found in the folder
+    for f_name in [
+        "structural_analysis_data.npz",
+        "energy_replica_state.npz",
+        "simulation_real_time_analysis",
+        "info.yaml"
+    ]:
+        if not results_dir.joinpath(f_name).exists():
+            print(f"Can't find cleaned results files for {f_name}")
```

Should this maybe throw an error instead?